### ISSUE TITLE
bugs. Too many open files

### ISSUE THREAD

**Issue by cxwx** [VidocqH/lsp-lens.nvim#41]
when using nvim-lspconfig clangd, lsp-lens together
go to definition
jump to another file after lspconfig jump to the right one.
then press <C-o> to go back,
cause.
```
Error detected while processing BufEnter Autocommands for "*"..BufReadPost Autocommands for
 "*":
Error executing lua callback: ...brew/Cellar/neovim/0.9.5/share/nvim/runtime/filetype.lua:2
4: Error executing lua: ...brew/Cellar/neovim/0.9.5/share/nvim/runtime/filetype.lua:25: Buf
Enter Autocommands for "*"..BufReadPost Autocommands for "*"..FileType Autocommands for "*"
..function <SNR>1_LoadFTPlugin[19]..script /Users/chenxu/cx_rc/config/nvim/after/ftplugin/c
pp.lua: Vim(runtime):E5112: Error while creating lua chunk: cannot open 
~/.config/nvim/after/ftplugin/cpp.lua: Too many open files
stack traceback:
        [C]: in function 'nvim_cmd'
        ...brew/Cellar/neovim/0.9.5/share/nvim/runtime/filetype.lua:25: in function <...bre
w/Cellar/neovim/0.9.5/share/nvim/runtime/filetype.lua:24>
        [C]: in function 'nvim_buf_call'
        ...brew/Cellar/neovim/0.9.5/share/nvim/runtime/filetype.lua:24: in function <...bre
w/Cellar/neovim/0.9.5/share/nvim/runtime/filetype.lua:10>
stack traceback:
        [C]: in function 'nvim_buf_call'
        ...brew/Cellar/neovim/0.9.5/share/nvim/runtime/filetype.lua:24: in function <...bre
w/Cellar/neovim/0.9.5/share/nvim/runtime/filetype.lua:10>
```

remove lsp-lens, lspconfig works fine

Is anyone has the same issues?

**Comment by mortymacs** [comment]
What's your current `limits`?
I'm uncertain if it pertains specifically to this plugin.
I encountered similar issues with other plugins, such as git-integrated, but resolved them by increasing my `limits` in my NixOS config: https://github.com/mortymacs/dotfiles/blob/main/nixos/common/misc.nix#L12

**Comment by kyu08** [comment]
I'v got a error like `Process failed to start: too many open files: "/opt/homebrew/bin/git"` too in few days ago.
In my macOS environment, following instruction fixed that error.

1. `sudo vi /Library/LaunchDaemons/limit.maxfiles.plist`.
2. Write following code to the files.
    ```xml
    <?xml version="1.0" encoding="UTF-8"?>
    <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
    <plist version="1.0">
      <dict>
        <key>Label</key>
        <string>limit.maxfiles</string>
        <key>ProgramArguments</key>
        <array>
          <string>launchctl</string>
          <string>limit</string>
          <string>maxfiles</string>
          <string>524288</string>
          <string>524288</string>
        </array>
        <key>RunAtLoad</key>
        <true/>
        <key>ServiceIPC</key>
        <false/>
      </dict>
    </plist>
    ```

3. Reboot your mac.

**Comment by YodaEmbedding** [comment]
In the settings, I disabled:

```lua
sections.git_authors = false
```

Changing limits wasn't really working for me.

**Comment by YodaEmbedding** [comment]
The code doesn't look too suspicious:

https://github.com/VidocqH/lsp-lens.nvim/blob/48bb1a7e271424c15f3d588d54adc9b7c319d977/lua/lsp-lens/lens-util.lua#L194-L197

Though it's missing a `"--", ` before `file_path` to correctly handle files that start with a `-`, and also not be susceptible to incredibly common dotfiles injection attacks. :)

Maybe `git blame` is called too many times at once? Perhaps git fails with simultaneous calls. Or maybe it gets stuck on one of them, and then just repeatedly keeps calling `git blame` on that same file. If `git blame` freezes or whatnot, maybe the plugin times that command out after 500ms, and then calls `git blame` again while the previous `git blame`(s) are still running. Not sure. Maybe this needs to be logged with timestamps.

Also, I'm not sure how it deals with changes to files that have not been saved yet. I may be wrong, but perhaps these errors seem to occur more frequently when I'm adding new classes or methods?

**Comment by worming004** [comment]
I encounter similar bug. When opening huge golang file(33k lines), nvim process goes from 18 open file descriptor to reach my limit of 1024

By just deactivating lsp-lens, nvim goes from 18 to 21 files opened

Counted with `ls /proc/<pid of nvim--embed>/fd | wc -l`

**Comment by serpro69** [comment]
Same issue, spent so many hours trying to find the culprit, until I finally discovered that lsp-lens was the culprit. Had to disable this plugin altogether because nvim was pretty much unusable otherwise since it would start spitting out 'too many open files' sooner or later, and on mac it was even worse since default limits are quite low. It was happening with all languages: go, ts/js, even things like terraform.
Good idea for the plugin, but things like these need to be fixed before it can be used.

**Comment by worming004** [comment]
@serpro69, read https://github.com/VidocqH/lsp-lens.nvim/pull/48#issuecomment-2892638209 FYI.

**Comment by serpro69** [comment]
> [@serpro69](https://github.com/serpro69), read [#48 (comment)](https://github.com/VidocqH/lsp-lens.nvim/pull/48#issuecomment-2892638209) FYI.

Yeah, I figured that it's not maintained since last commit was 2 years ago.
It's really not that big of a deal for me, the less plugins - the better 😁 It was just annoying to debug this issue, so I thought I'd leave a comment for anyone else who might stumble onto this.

**Comment by cxwx** [comment]
close as too old
